.github/workflows/build-user-config.yml

```diff
@@ -0,0 +1,96 @@
+name: Reusable user config build
+
+on: 
+  workflow_call:
```

Comment:
https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onworkflow_callinputs indicates that we can add inputs. A couple of inputs that I think would be useful:

```yaml
on:
  workflow_call:
    inputs:
      build_matrix_path:
        description: Path to the build matrix file
        default: build.yaml
        required: false
        type: string
      config_path:
        description: Path to the config directory
        default: config
        required: false
        type: string
```
This would give people some flexibility as to the organization of their repo. 

For example, in https://github.com/joelspadin/NumBLE/tree/zephyr-3.0 I put both the hardware design files for a keyboard and the ZMK config for it in the same repo. To make it easier to understand, I renamed "config" to "zmk-config" for that repo. It would be great if I could just call this job with

```yaml
with:
  config_path: zmk-config
```

Reply:
Yes, that makes sense. I will include these adjustments in a commit this afternoon.

If we are already adding functionality in this PR, I would also add an input for the chosen artifact file type. For example, I want the `*.bin` file for flashing with `dfu-util`, instead of  `*.hex`. Why is it `*.hex` in the first place?

Reply:
Added changes.